import_handlers.js

```diff
@@ -0,0 +1,154 @@
+        this.date = date;
+        this.from = from;
+        this.to = to;
+        this.narrative = narrative;
+        this.amount = parseFloat(amount);    
+    }
+}
+
+class Account {
+    constructor(name) {
+        this.name = name;
+        this.transactionRefs = [];
+    }
+
+    calcBalance() {
+        var running_total = 0;
+        let account_name = this.name;
+        this.transactionRefs.forEach(function(t) {
+            if (t.to === account_name) {
+                running_total += t.amount;
+            } else {
+                running_total -= t.amount;
+            }
+        });
+        return running_total;
+    }
+}
+
+function import_csv_data(inputPath, accounts) {
+    let transactions = [];
+
+    const csv_out = fs.readFileSync(inputPath, 'utf8');
+    const lineSplit = csv_out.split('\n');
+
+    for (var i=1; i<lineSplit.length-1; i++) {
+        // Start at row 1, as row 0 contains the column headings
+        // End one before the final element, since the last array item is a blank line
+        const transaction = lineSplit[i].split(',');
+
+        // Extract the fields
+        const date = transaction[0];
+        const from = transaction[1];
+        const to = transaction[2];
+        const narrative = transaction[3];
+        const amount = transaction[4];
+        
+        // Check valid amounts (with isNaN)
+        if (isNaN(parseFloat(amount))) {
+            console.log('Not a valid amount');
+            continue;
+        }
+        
+        // Valid date checking
+        const date_moment = moment(date, 'DD/MM/YYYY');
+        if (!date_moment.isValid()) {
+            console.log('Not a valid date:', date);
+            continue
+        }
+
+        // Create transaction objects
+        transactions.push(new Transaction(date_moment, from, to, narrative, amount));
+    }
+
+    // Use the transactions just loaded to update the accounts
+    updateAccounts(accounts, transactions);
+    return transactions;
+}
+
+function import_json_data(inputPath, accounts) {
+    let transactions = [];
+
+    const json_str_out = fs.readFileSync(inputPath, 'utf8');
+    const json = JSON.parse(json_str_out);
+
+    json.forEach(function (t) {
+        const date_moment = moment(t['Date'], 'YYYY-MM-DD');
+        if (!date_moment.isValid()) {
+            console.log('Not a valid date', date_moment);
+            return;
+        }
+
+        transactions.push(new Transaction(date_moment,
+                                          t["FromAccount"],
+                                          t["ToAccount"],
+                                          t["Narrative"],
+                                          t["Amount"]));
+    });
+
+    updateAccounts(accounts, transactions);
+    return transactions;
+}
+
+function import_xml_data(inputPath, accounts) {
+    let transactions = [];
+
+    const xml_str_out = fs.readFileSync(inputPath, 'utf8');
+    var parseString = xml2js.parseString;

```

Comment:
1) I actually think directly using `xml2js.parseString(...)` is slightly clearer than splitting it over two lines.
2) This can *definitely* be a `const`.